cpp/src/kmeans/kmeans-inl.cuh

```diff
@@ -529,6 +529,11 @@ DataT *KMeans<DataT, IndexT>::centroids() {
   return _centroidsRawData.data();
 }
 
+template <typename DataT, typename IndexT>
+const double *KMeans<DataT, IndexT>::getInertia() {
+  return &inertia;
```

Comment:
It's very strange returning a pointer in this way, because the kmeans object owns this pointer. I would return the value here and use the `&` reference operator in the call to `memcpy`